scanner/src/test/kotlin/storages/ClearlyDefinedStorageTest.kt

```diff
@@ -0,0 +1,405 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.storages
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import com.github.tomakehurst.wiremock.client.WireMock.anyUrl
+import com.github.tomakehurst.wiremock.client.WireMock.equalTo
+import com.github.tomakehurst.wiremock.client.WireMock.get
+import com.github.tomakehurst.wiremock.client.WireMock.stubFor
+import com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+
+import io.kotest.assertions.fail
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.ints.shouldBeLessThan
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+
+import java.net.ServerSocket
+import java.time.Duration
+import java.time.Instant
+
+import org.ossreviewtoolkit.clearlydefined.ComponentType
+import org.ossreviewtoolkit.model.Failure
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Result
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScanResultContainer
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.Success
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.ClearlyDefinedStorageConfiguration
+
+private const val PACKAGE_TYPE = "Maven"
+private const val NAMESPACE = "someNamespace"
+private const val NAME = "somePackage"
+private const val VERSION = "0.1.8"
+private const val COMMIT = "02b7f3d06fcbbedb44563aaa88ab62db3669946e"
+private const val SCANCODE_VERSION = "3.2.2"
+
+/** The name of the file with the test results from ClearlyDefined. */
+private const val RESULT_FILE = "clearlydefined_scancode.json"
+
+/** A delta for comparing timestamps against the current time. */
+private val MAX_TIME_DELTA = Duration.ofSeconds(30)
+
+/** The ClearlyDefined URL referencing the test package. */
+private const val PACKAGE_URL = "maven/mavencentral/$NAMESPACE/$NAME/$VERSION"
+
+/** Path to a file contained in the test ClearlyDefined result. */
+private const val TEST_PATH =
+    "src/main/java/org/apache/commons/configuration2/tree/DefaultExpressionEngine.java"
+
+/** A test identifier. */
```

Comment:
Nit: I think it's fine to not overdo it and omit comments that basically just repeat the variable name.